app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt

```diff
@@ -34,6 +39,17 @@ internal class InitializationActivity : AppCompatActivity() {
     private lateinit var settings: Settings
     private var splashScreenActive = true
 
+    @RequiresApi(Build.VERSION_CODES.S)
+    private val activityResultLauncher =
+        registerForActivityResult(StartActivityForResult()) {
+            val manager = ContextCompat.getSystemService(this, AlarmManager::class.java)
+            if (manager?.canScheduleExactAlarms() == true) {
+                tryAuthenticate()
+            } else {
+                alarmDialog()
+            }
```

Comment:
Can this part be extracted to another separate method and reused on the lines 75+?
The checks are a little different == false and == true, but it looks doable.

Reply:
Doable